standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -803,6 +805,22 @@ public Partition alterPartition(RawStore msdb, Warehouse wh, String catName, Str
     return oldParts;
   }
 
+  private void checkTableTypeConversion(Database db, Table oldTbl, Table newTbl)
+          throws InvalidOperationException {
+    // If the given DB is enabled for replication and strict managed is false, then table type cannot be changed.
+    // This is to avoid migration scenarios which causes ACID table to be converted to external at replica.
+    // As ACID tables cannot be converted to external table and vice versa, we need to restrict this conversion at
+    // primary as well.
```

Comment:
Is every managed table going to be converted to an ACID table on the replica? If not, should we just restrict those managed tables which will be converted?

Reply:
duplicate comment.